### DISCUSSION TITLE
Wake center of the curl model in FAST Farm

### DISCUSSION THREAD

**Discussion by MYMahfouz** [OpenFAST/openfast#1778] [Q&A]
Dear all,

I want to know the wake center from the curl model output in FAST.Farm.

Can I use the output channel `WkPosZTNDN` for the height of the wake center?

I can see from the results that the height of the wake center is lower than the hub height of the turbine. I was expecting the wake center to be higher than the hub height because of the wake deflection. 

The turbine I am simulating has a hub height of `135m`, the results from `WkPosZTNDN` channel gives around `134.1m` for a fixed bottom wind turbine and `134.5m` for a floating turbine. These values are at a distance `4D` behind the turbine.

If this is not the correct channel what channel can I use?

Thank you.

Best regards,
Youssef

**Comment by jjonkman** [comment]
Dear @MYMahfouz,

FAST.Farm outputs WkPosδTαDγ provide the (δ = X,Y,Z) coordinates of the center of a wake plane for the polar, curl, and Cartesian wake models.  With the curled model, however, the wake center is not necessarily at the center of the wake plane due to its kidney-like shape.  To calculate the actual center of a curled wake, it is better to post-process the wake center via a post-processing tool such as SAMWHICH: https://ewquon.github.io/waketracking/.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

Thank you for your reply.

I have tried to use SAMWICH as you suggested and I followed the example [here](https://github.com/ewquon/waketracking/blob/master/examples/example_00_quick_start.ipynb).

It doesn't seem to be working as expected for all cases.

I have attached an example for the results I am getting at 1D behind the turbine.

Are there other examples or other methods for this calculation?

Thanks again for all the help.

Best regards

![image](https://github.com/OpenFAST/openfast/assets/33868271/77902777-765f-43d3-8684-38ef22f93e06)

**Reply by jjonkman** [reply]
Dear @MYMahfouz,

Can you clarify what is not working?

That said, I've not used SAMWHICH myself, so, I'm not one that can provide much support.  Perhaps others can comment or you can post your questions in the SAMWHICH repository.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

The result in the figure shows wake center at 112m height (below the hub center at 135m) and the contour doesn't cover the entire wake. This is why I am assuming I am doing something wrong or it isn't working as expected.

Are there other libraries/tools which you are aware of which I can try?

I will also post my question on the SAMWICH repo as you suggested.

Thanks for the help and the quick and useful reply.

Best regards

**Reply by jjonkman** [reply]
Dear @MYMahfouz,

SAMWHICH provides multiple ways of "fitting" the wake.

I'm not aware of other libraries/tools similar to SAMWHICH.

Best regards.

**Comment by Marcodep23** [comment]
Dear @jjonkman,

I’m experiencing the same issue reported by @MYMahfouz. I encountered the same problem when simulating the IEA 15 MW + VolturnUS configuration.

I ran a turbulent inflow case with a hub-height mean wind speed of 10 m/s (no peak shaving) to evaluate the effect of vertical deflection at the maximum misalignment angle. The average platform pitch is approximately 4.5°, in addition to the 6° nominal shaft tilt.

I compared the time-averaged VTK results with a bottom-fixed reference case. I expected to observe results similar to those presented in https://wes.copernicus.org/articles/9/1827/2024/wes-9-1827-2024.html, showing a clearly visible vertically deflected wake. However, my results look as follows:
  
<img width="599" height="241" alt="image" src="https://github.com/user-attachments/assets/35b8f5a5-a938-4b87-aeb3-b024877d2320" />

Before running the simulations, I ensured that the model was properly calibrated for the 15 MW turbine, based on both forum discussions and relevant literature. I used the Curl Model with the following parameters:

fc = 0.007 Hz (equal to surge natural frequency)
k_VortexDecay = 0
k_vCurl = 5
C_Meander = 2.1 (though this should be ineffective for the Curl model)
A low-domain grid size following your recommendations

Do you have any suggestions or insights on why the floating case does not capture the expected vertically deflected wake?

Best regards,

**Reply by jjonkman** [reply]
Dear @Marcodep23,

Presumably you have used DEFAULT values for the other curled-wake parameters in FAST.Farm?

Presumably you are plotting a time-averaged Y-Z slice (not X-Y as stated) at 6.75D downstream of the turbine; do you see any vertical deflection if you look at a X-Z slice?

Can you run an equivalent case with steady inflow (without turbulence) to better see the curled wake shape, not smeared from wake meandering?

Best regards,

**Reply by Marcodep23** [reply]
Dear @jjonkman,

Thank you for your quick reply.

I only modified the parameters mentioned, as from my understanding, those are typically used for model calibration (mainly based on [WES 2024](https://wes.copernicus.org/articles/9/1827/2024/wes-9-1827-2024.html)
 and [WES 2025 Discussion](https://wes.copernicus.org/articles/10/1033/2025/wes-10-1033-2025-discussion.html)
). Are there other parameters that are also useful for calibrating the model? In the literature, I could only find references to these ones.

You are correct — that was a typo; it should have been a Y–Z slice. Below you can find the X–Z plane as requested. A very small deflection can be observed, but it is almost negligible.
<img width="815" height="352" alt="image" src="https://github.com/user-attachments/assets/0a03c90c-d678-408d-acd3-921e2f0fc662" />

Regarding the steady inflow, I can certainly run that case. However, the case shown here already has a TI = 2%, so its influence should be quite limited.

Best regards,
Marco

**Reply by jjonkman** [reply]
Dear @Marcodep23,

I was assuming you are using DEFAULT values for the other curled wake parameters, but wanted to confirm.

I'll hold off until commenting further until you share results with steady inflow, but I'm not really seeing the curled wake shape that should induce the wake deflection, which is odd given the 10deg tilt.

Best regards,

**Reply by Marcodep23** [reply]
Dear @jjonkman,

I ran two additional simulations as you requested, further simplifying the problem. In both cases, the turbines have all ElastoDyn DOFs disabled. The rotor speed is fixed at 7.13 rpm (interpolated from the steady-state curve at a wind speed of 10 m/s), and the blade pitch is set to zero.

The only difference between the two cases is that the “Floating” turbine has an initial platform pitch angle of 4.5°, while the “Bottom-Fixed” one does not. The inflow is uniform and steady at 10 m/s. The wake model parameters are the same as previously mentioned; all others remain at their default values.

The goal of this setup was to isolate and observe the vertical wake deflection induced by the platform pitch displacement.

Here are the results:
<img width="833" height="251" alt="image" src="https://github.com/user-attachments/assets/16ca054d-7a8d-47e3-a341-6513eaff9c30" />
<img width="654" height="376" alt="image" src="https://github.com/user-attachments/assets/d53fb731-94b9-430f-bb84-743d9f58055c" />

From these results, it appears that—with the calibrated values from literature—the Curl model does not capture the vertical wake deflection for the IEA 15 MW turbine. This is somewhat unexpected, since the same model correctly reproduces both horizontal wake steering and vertical deflection in tests with the NREL 5 MW turbine. However, in my case study, this behavior is not observed. 
Do you have any insights into why this might be happening? Would it make sense to use the Polar model instead?

Best regards,
Marco

**Reply by jjonkman** [reply]
Dear @Marcodep23,

Thanks for running these cases.  In both cases, I still see a fairly strong horizontal wake deflection, which I presume is coming from from the wake rotation due to `Swirl` = TRUE (that is DEFAULT).  I'm wondering if what is happening is that the wake is curling up from the tilt/pitch, but with wake rotation, the kidney shape is rotating such that the vertical deflection actual becomes a horizontal deflection.  Can you run the same cases with `Swirl` = FALSE to confirm?

Best regards,
